tests/crashes/132335.rs

```diff
@@ -1,5 +1,10 @@
+// Regression test for #132335. This previously ICE'd due to ambiguity
+// in MIR typeck.
+
 //@ known-bug: #132335
-//@ compile-flags: -Znext-solver=globally --crate-type lib --edition=2018
+//@ compile-flags: -Znext-solver=globally --crate-type lib
+//@ edition: 2018
+//@ check-pass
```

Comment:
this was not part of the backported commit, meaning 1.86 should not have `check-pass`

Reply:
I... I have no idea.